### DISCUSSION TITLE
How does react-flow give an experience of infinite space ?

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#3245] [General]
When I create two nodes connected to one another, and I click and drag one node towards left, the background space seems to be growing. 

Absolute position the div, as mouse move left,, then add 500px of left side size?

**Comment by bcakmakoglu** [comment]
There's no "size" added or anything in the viewport (you can open the Devtools in your Browser and check).
The elements are positioned absolutely, that is true, but the panning and zooming is done via the `transform` css property and not by adding some size to any element in the dom.

This is made possible with the help of [d3-zoom](https://d3js.org/d3-zoom).
